Code/Framework/AzCore/AzCore/std/hash_table.h

```diff
@@ -592,7 +592,7 @@ namespace AZStd
             // insert [first, last) one at a time
             for (; first != last; ++first)
             {
-                const key_type& valueKey = Traits::key_from_value(*first);
+                const key_type valueKey = Traits::key_from_value(*first);
```

Comment:
This can potentially make an expensive copy of the key, when it does return a reference, such as when it is an AZStd::string.

Instead of storing the key in a local variable, we should just perform the `operator*()` when it is actually needed in the call to `m_hasher` and `find_insert_position`

Reply:
This is fixed in the latest revision